### ISSUE TITLE
Notify exactly one Pull-Request

### ISSUE THREAD

**Issue by devalru** [jenkinsci/stashnotifier-plugin#151]
Hi all,

in my view there is serious problem with the notification based on commits.
BitBucket ( v4.12.0) updates *all* pull-requests with the same source branch, because the plugin sends the commit as identifier and not the pull-request ID / pull-request URL.

Example:
If the build for pull-request A->B is successfull than all open pull-requests A->X in BitBucket
are updated (with the same status) because of the common commit in branch A.
 
Do you have any suggestions ?
Thank you !

**Comment by scaytrase** [comment]
This is not the problem of plugin, but of your pull-request builds process.

You are correct saying that 

A->B and A->X are updated, because builds are done against last commit of branch A, which is same for all pull request. And the status indicates the build state for commit, not the branch, so both PR get green builds.

That's how Bitbucket works, not this plugin.

To overcome this problem I suggest you to overhaul the PR building process:

1. Use [Bruho Build Status](https://marketplace.atlassian.com/plugins/com.bolyuba.stash.plugin.stash-build-status-plugin/server/overview) plugin
2. Configure it to require automerge builds to pass
3. Configure jenkins to build against [shadow branches](http://docs.bruho.com/atlas/stash/addons/buildstatus/#solution) instead (or with) 

In short, Bitbucket creates a shadow branch `/refs/pull-requests/<id of your pull request>/merge` for each pull-request you have, if they don't have a merge conflict. This branch represents the merged state of PR if it gets merged as is.

You can run a build against this state to be sure that merged state is correct according to your rules (tests, style, deploy, etc)
